pkg/scheduler/scheduler.go

```diff
@@ -149,14 +190,19 @@ func (s *Scheduler) schedule(ctx context.Context) {
 			//      evaluated in the same cycle is not taken into account.
 			//   2. An already admitted workload from a different cluster queue is borrowing, since all workloads
 			//      evaluated in the current cycle will compete for the resources that are not borrowed.
-			if usedCohorts.Has(cq.Cohort.Name) && (e.assignment.Borrows() || cq.Cohort.HasBorrowingQueues()) {
-				e.status = skipped
-				e.inadmissibleMsg = "other workloads in the cohort were prioritized"
-				continue
+			if cycleCohortsUsage.hasConflictWith(cq.Cohort.Name, e.assignment.Usage) && (e.assignment.Borrows() || cq.Cohort.HasBorrowingQueues()) {
```

Comment:
Is this `&& (e.assignment.Borrows() || cq.Cohort.HasBorrowingQueues()) ` still required actually? IIUC the comment above this is meant to prevent two workloads admitted if we have borrowing to prevent overcommitment. However, as we have a dedicated check for overcommitment, wondering what are scenarios when this is still needed.

Reply:
Yes, if there is no borrowing going on and therefore every queue is assigning in its nominal quota there is no reason to any extra checks.

Reply:
IIUC, if there is no borrowing, then additional checks are not needed. Now, when there is borrowing then we have the new capacity check, which should also work fine for the non-borrowing case. 

So, I'm considering if the code was still correct if we removed `&& (e.assignment.Borrows() || cq.Cohort.HasBorrowingQueues()) `, I cannot find a problematic scenario, but maybe I'm missing something.

Reply:
And if we dropped, `&& (e.assignment.Borrows() || cq.Cohort.HasBorrowingQueues())` then I would consider wrapping invocations of `hasCommonResources`, `totalUsageForCommonResources` under `CanFit` that would do all the checks.

Reply:
`(e.assignment.Borrows() || cq.Cohort.HasBorrowingQueues()` provide a fast way out, without the need of extracting the common flavor/resorces used and check the fit.

Reply:
moved to https://github.com/kubernetes-sigs/kueue/pull/1039#discussion_r1288099746